From 3e24ddd80b2a2938a132a9df97038ecc0410a390 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Wed, 27 Nov 2024 06:47:11 +0000
Subject: [PATCH] install: Make sure the default driver is the in-tree driver

We put the out of tree drivers *.ko to weak-updates folder to make
sure the in-tree drivers will be loaded by default in the original
design. But we found the default drivers will be the OOT drivers
in some situation, which is not that we want.

The issue is same with https://access.redhat.com/solutions/3536351.
To fix the issue, we run "depmod -a" before load the drivers.

Verification:
1. There is not out-of-tree-drivers in the boot cmmandline. Make
   sure the ice, iavf and i40e will be loaded with the in-tree
   drivers.
2. out-of-tree-drivers=ice, make sure the iavf and i40e will be
   loaded with the in-tree driver.
3. out-of-tree-drivers=ice,iavf,i40e make sure the three drivers
   will be loaded with the OOT drivers.

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 init-ostree-install.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init-ostree-install.sh b/init-ostree-install.sh
index 0d047fb..712e65a 100644
--- a/init-ostree-install.sh
+++ b/init-ostree-install.sh
@@ -479,6 +479,7 @@ do_mount_fs() {
 
 network_modules() {
 	ln -s /lib/firmware/updates/intel/ice/ddp/ice-*.pkg /lib/firmware/intel/ice/ddp/ice.pkg
+	depmod -a
 	if grep -s -q '\sout-of-tree-drivers=' /proc/cmdline ; then
 		drivers=(`echo ${OOT_DRIVERS} | tr ',' ' '` )
 		for driver in ${drivers[@]}
-- 
2.47.0

